GOVERNANCE.md

```diff
@@ -107,6 +107,29 @@ Lazy consensus does _not_ apply to the process of:
 
 * Removal of maintainers from Velero
 
+## Deprecation Policy
+
+### Deprecation Process
+
+Any contributor may introduce a request to deprecate a feature or an option of a feature by opening a feature request issue in the vmware-tanzu/velero GitHub project. The issue should describe why the feature is no longer needed or has become detrimental to Velero, as well as whether and how it has been superseded. The submitter should give as much detail as possible.
+
+Once the issue is filed, a discussion period begins which ends at the beginning of the second community meeting after the opening of the issue, and is held at the community meeting and on the issue itself; the person who opens the issue or a maintainer should add that date and time in a comment in the issue as soon after the issue is opened as possible.
+
+The feature will be deprecated by a supermajority vote of 50% plus one of the project maintainers at the time of the vote tallying, which is 72 hours after the end of the community meeting that is the end of the comment period. (Maintainers are permitted to vote in advance of the deadline, but should hold their votes until as close as possible to hear all possible discussion.) Votes will be tallied in comments on the issue. 
+
+Non-maintainers may add non-binding votes in comments to the issue as well; these are opinions to be taken into consideration by maintainers, but they do not count as votes. 
+
+If the vote passes, the deprecation window takes effect in the subsequent release, and the removal follows the schedule. 
+
+### Schedule
+If depreciation proposal passes by supermajority votes, the feature is deprecated in the next minor release and the feature can be removed completely after two minor version or equivalent major version e.g., if feature gets deprecated in Nth minor version, then feature can be removed after N+2 minor version or its equivalent if the major version number changes.
+
+### Deprecation Window
+
+The deprecation window is the period from the release in which the deprecation takes effect through the release in which the feature is removed. During this period, only critical security vulnerabilities and catastrophic bugs should be fixed.
+
+**Note:** If a deprecated feature must remain available for general use for two releases after deprecation, perhaps a backup that uses this deprecated feature must be fully-supported for restore for four releases.
```

Comment:
"must be fully supported for restore for four releases" perhaps we want to keep the code for backup but only allow it for generating and running test cases, not general use by end user. 

Running more than one version to test would be ideal but more difficult imo especially in unit testing.

Reply:
what use case does the "*Note:**" section try to solve? thanks.

Reply:
@renmaosheng 
See [this comment ](https://github.com/vmware-tanzu/velero/pull/5532#discussion_r1170938898), if a feature is essential to the backup which for example, impacts the backed up data, when we deprecate the feature, we should guarantee that users are still able to access their backups, e.g., restore from them.
Here Restic deprecation is right an example, users may have some previous backups created by Restic, and they may want to preserve the backups for quite long time, remaining able to restore from them.

Reply:
Restic is the main example we had in mind. If we deprecate Restic in 1.13, then restic must be available for *new* backups for N+2 releases, but prior restic backups must be available for restore for longer. Otherwise, if we removed Restic completely in 1.15 (rather than just on the backup side), this would mean that a user could make a Restic backup in 1.14, upgrade to 1.15 and that "one release old" backup would no longer be restorable.

Reply:
I also find the sentence 
```
perhaps a backup that uses this deprecated feature must be fully-supported for restore for four releases.
```
is a little confusing.

How about we just explicitly mention that a rule of thumb for velero is to be able to restore a backup that is created in version `n-2`, and rule must not be violated when a deprecated feature is removed from velero?

Reply:
cc @weshayutin

Reply:
@reasonerjt @weshayutin Something like this?
"If a backup relies on a deprecated feature, then backups made with the last Velero release before this feature is removed must still be restorable in version `n+2`. For something like restic support, that might mean that restic is removed from the list of supported uploader types in version `n` but the underlying implementation required to restore from a restic backup won't be removed until release `n+2`"

Reply:
+1 @sseago 
@reasonerjt @weshayutin I can update the PR if we agree upon what @sseago suggested here.

Reply:
++ I agree w/ what @sseago said there

Reply:
@shubham-pampattiwar 
Sorry for the late response, the proposed wording by @sseago looks good to me.

Reply:
@reasonerjt @weshayutin @sseago Done, updated the PR.